test/org/sagebionetworks/bridge/services/SchedulePlanServiceMockTest.java

```diff
@@ -0,0 +1,123 @@
+package org.sagebionetworks.bridge.services;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotNull;
+
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.sagebionetworks.bridge.TestUtils;
+import org.sagebionetworks.bridge.dao.SchedulePlanDao;
+import org.sagebionetworks.bridge.dynamodb.DynamoSchedulePlan;
+import org.sagebionetworks.bridge.models.schedules.Activity;
+import org.sagebionetworks.bridge.models.schedules.Schedule;
+import org.sagebionetworks.bridge.models.schedules.SchedulePlan;
+import org.sagebionetworks.bridge.models.schedules.ScheduleType;
+import org.sagebionetworks.bridge.models.schedules.SimpleScheduleStrategy;
+import org.sagebionetworks.bridge.models.surveys.Survey;
+import org.sagebionetworks.bridge.validators.SchedulePlanValidator;
+
+public class SchedulePlanServiceMockTest {
+
+    private String surveyGuid;
+    private SchedulePlanServiceImpl service;
+    
+    private SchedulePlanDao mockSchedulePlanDao;
+    private SurveyService mockSurveyService;
+    
+    @Before
+    public void before() {
+        mockSchedulePlanDao = mock(SchedulePlanDao.class);
+        mockSurveyService = mock(SurveyService.class);
+        
+        service = new SchedulePlanServiceImpl();
+        service.setSchedulePlanDao(mockSchedulePlanDao);
+        service.setSurveyService(mockSurveyService);
+        service.setValidator(new SchedulePlanValidator());
+        
+        Survey survey = TestUtils.getSurvey(false);
+        when(mockSurveyService.getSurveyMostRecentlyPublishedVersion(any(), any())).thenReturn(survey);
+        surveyGuid = survey.getGuid();
+    }
+    
+    @Test
+    public void surveyReferenceIdentifierFilledOutOnCreate() {
+        SchedulePlan plan = createSchedulePlan();
+        
+        ArgumentCaptor<SchedulePlan> spCaptor = ArgumentCaptor.forClass(SchedulePlan.class);
+        
+        service.createSchedulePlan(plan);
+        verify(mockSurveyService).getSurveyMostRecentlyPublishedVersion(any(), any());
+        verify(mockSchedulePlanDao).createSchedulePlan(spCaptor.capture());
+        
+        String identifier = spCaptor.getValue().getStrategy().getAllPossibleSchedules().get(0).getActivities().get(0)
+                .getSurvey().getIdentifier();
+        assertNotNull(identifier);
```

Comment:
Would it be better to explicitly set the survey identifier to some test value, so you can string match it here?